src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/add_sub_menu_request.cc

```diff
@@ -96,14 +96,23 @@ void AddSubMenuRequest::Run() {
     return;
   }
 
-  const std::string& menu_name =
-      received_msg_params[strings::menu_name].asString();
-
   const uint32_t parent_id =
       received_msg_params.keyExists(strings::parent_id)
           ? received_msg_params[strings::parent_id].asUInt()
           : 0;
 
+  if (0 != parent_id) {
+    smart_objects::SmartObject parent = app->FindSubMenu(parent_id);
+    if (smart_objects::SmartType_Null == parent.getType()) {
+      SDL_LOG_ERROR("Parent ID " << parent_id << " doesn't exist");
+      SendResponse(false, mobile_apis::Result::INVALID_ID);
```

Comment:
did you consider adding a message to this response to explain why invalid id was returned?

Reply:
@iCollin Reason is added in 017d900f74ba6e8c573e13dbed257d437af70312